### PR TITLE
Improve documentation for FeatureEnvy and UtilityFunction.

### PR DESCRIPTION
**Pull request by troessner**

### CODE REVIEW

File: docs/Feature-Envy.md [hunk 1]
```diff
@@ -2,28 +2,48 @@
...
+`Feature Envy` often arises because it must manipulate other objects (usually its arguments) to get them into a useful form, and one force preventing them (the arguments) doing this themselves is that the common knowledge lives outside the arguments, or the arguments are of too basic a type to justify extending that type. Therefore there must be something which 'knows' about the contents or purposes of the arguments.  That thing would have to be more than just a basic type, because the basic types are either containers which don't know about their contents, or they are single objects which can't capture their relationship with their fellows of the same type. So, this thing with the extra knowledge should be reified into a class, and the utility method will most likely belong there.
+
+## Example
+
+Running reek on:
 
 ```Ruby
-class Cart
-  def price
-    @item.price + @item.tax
```
Comment (troessner) [inline docs/Feature-Envy.md]: The example before would actually NOT report FeatureEnvy.:)

File: docs/Feature-Envy.md [hunk 2]
```diff
@@ -2,28 +2,48 @@
...
+(item.price - item.rebate)
+```
+
+belongs to the Item class, not the Warehouse.
 
 ## Current Support in reek
 
 `Feature Envy` reports any method that refers to self less often than it refers to (ie. send messages to) some other object.
 
+## Differences to `Utility Function`
```
Comment (troessner) [inline docs/Feature-Envy.md]: This is duplicated in the UtilityFunction docs below but I believe it is so important to make this clear that this is ok.

File: docs/Utility-Function.md [hunk 3]
```diff
@@ -2,11 +2,9 @@
...
-
-A Utility Function often arises because it must manipulate other objects (usually its arguments) to get them into a useful form, and one force preventing them (the arguments) doing this themselves is that the common knowledge lives outside the arguments, or the arguments are of too basic a type to justify extending that type. Therefore there must be something which 'knows' about the contents or purposes of the arguments.  That thing would have to be more than just a basic type, because the basic types are either containers which don't know about their contents, or they are single objects which can't capture their relationship with their fellows of the same type. So, this thing with the extra knowledge should be reified into a class, and the utility method will most likely belong there.
+`Utility Function` is heavily related to [Feature Envy](Feature-Envy.md), please see check out the explanation there why `Utility Function` is something you should care about.
```
Comment (troessner) [inline docs/Utility-Function.md]: The introduction for both was an exact duplication.

File: docs/Utility-Function.md [hunk 4]
```diff
@@ -37,10 +35,10 @@ test.rb -- 2 warnings:
...
+## Differences to `Feature Envy`
 
-`reek`'s `Utility Function` detector supports the [Basic Smell Options](Basic-Smell-Options.md), plus:
+`Feature Envy` is only triggered if there are some references to self and `Utility Function` is triggered if there are no references to self.
+
+## Configuration
 
-| Option         | Value       | Effect  |
-| ---------------|-------------|---------|
-| `max_helper_calls` |  integer | The maximum number of method calls to other objects allowed within a method. Defaults to 2. |
```
Comment (troessner) [inline docs/Utility-Function.md]: This option was removed in #452 but we forgot to update the docs.:)

File: docs/Utility-Function.md [hunk 5]
```diff
@@ -2,11 +2,9 @@
...
-
-A Utility Function often arises because it must manipulate other objects (usually its arguments) to get them into a useful form, and one force preventing them (the arguments) doing this themselves is that the common knowledge lives outside the arguments, or the arguments are of too basic a type to justify extending that type. Therefore there must be something which 'knows' about the contents or purposes of the arguments.  That thing would have to be more than just a basic type, because the basic types are either containers which don't know about their contents, or they are single objects which can't capture their relationship with their fellows of the same type. So, this thing with the extra knowledge should be reified into a class, and the utility method will most likely belong there.
+`Utility Function` is heavily related to [Feature Envy](Feature-Envy.md), please check out the explanation there why `Utility Function` is something you should care about.
```
Comment (troessner) [inline docs/Utility-Function.md]: The introduction for both was an exact duplication.

File: docs/Feature-Envy.md [hunk 6]
```diff
@@ -2,28 +2,48 @@
 
 ## Introduction
 
-Feature Envy occurs when a code fragment references another object more often than it references itself, or when several clients do the same series of manipulations on a particular type of object.
+`Feature Envy` occurs when a code fragment references another object more often than it references itself, or when several clients do the same series of manipulations on a particular type of object.
```
Comment (mvz) [inline docs/Feature-Envy.md]: I'm not sure about writing smell names in monospace font. Can we offset them some other way?

File: docs/Feature-Envy.md [hunk 7]
```diff
@@ -2,28 +2,48 @@
...
+
+belongs to the Item class, not the Warehouse.
 
 ## Current Support in reek
 
 `Feature Envy` reports any method that refers to self less often than it refers to (ie. send messages to) some other object.
 
+## Differences to `Utility Function`
+
+`Feature Envy` is only triggered if there are some references to self and `Feature Envy` is triggered if there are no references to self.
```
Comment (mvz) [inline docs/Feature-Envy.md]: The second 'Feature Envy' should be 'Utility Function'.

**Comment by mvz** [comment]
Good changes in general. See my two comments.

File: docs/Feature-Envy.md (hunk 6 above)
Comment (troessner) [inline docs/Feature-Envy.md]: Hmmm. How about italic?

Comment (troessner) [inline docs/Feature-Envy.md]: @mvz @chastell let's go with italic for smell names and things like that. objections?

Comment (mvz) [inline docs/Feature-Envy.md]: No objections. :+1:

**Comment by troessner** [comment]
@mvz addressed all comments!

**Pull request merged**